cpp/src/rolling/rolling_detail.cuh

```diff
@@ -895,6 +899,235 @@ struct rolling_window_launcher {
       stream,
       mr);
   }
+
+  template <typename PrecedingIter, typename FollowingIter>
+  std::unique_ptr<column> get_collect_list_offsets(column_view const& input,
```

Comment:
This function only applies to the `COLLECT` aggregation. That is why the name reflects `collect`.

Here's a thought: Should these new helper functions be moved to a new `cudf::detail::collect` namespace, so that its limited scope is clearer?

Reply:
For the moment, I have renamed the function to `create_collect_offsets()`, in line with `create_collect_null_mask()`, `create_collect_gather_map()`, etc.
I'm happy to rename it again if you'd prefer a different convention.